### ISSUE TITLE
decode and effects based on sound source

### ISSUE THREAD

**Issue by LucianTothezan** [LewisSmallwood/IoT-Logitech-Z906#5]
Hello again. I was wondering if it would be possible to make this emulator behave as the original controller: based on the type of the source sound to turn on automatically the decode option and disable the effects. To my Knowledge, if the input sound(digital) is detected as being DTS or Dolby Digital, the decode will kick in automatically, thus disabling the effects. Is this possible to replicate? At this moment i have to turn decode off and effects off, in order to have a proper sound. I am using one of the optical inputs mainly.

**Comment by 3limin4tor** [comment]
Hello everyone. I've taken a closer look and tried it out a bit (but not done yet), as well as read the user manual of the sound system. In doing so, I came across some interesting information. Here is the excerpt from the user manual:
![image](https://github.com/LewisSmallwood/IoT-Logitech-Z906/assets/5002404/fb9c9779-e93a-433e-8973-29cc534f097f)


If I interpret this correctly, digital inputs are only the COAX and the two optical inputs, at least this is what i interpret from the manual. See here:
![image](https://github.com/LewisSmallwood/IoT-Logitech-Z906/assets/5002404/632e4930-f6b5-42be-acbf-f427fe13f5c5)

However, if a stereo signal is received, one can choose how it should be played back (2.1, 4.1, or 3D, all satellites). According to my understanding, the effect setting can thus be applied to all inputs.

The behavior for the digital inputs would therefore be as follows: If a stereo signal is received, the decode display goes out and you can set how the stereo signal should be played back on the satellites. On the other hand, if a DTS or Dolby signal is received, the effect is automatically turned off and the decode activated (decode LED is then on). If you then switch back to a stereo signal, the decode LED goes out again and the last active effect is reactivated. However, I am not entirely sure if that is exactly how it is, this is merely a speculation. I will test this to be sure and share my findings here as a response.

@LewisSmallwood:
How do you get the command for decode on / off? Unfortunately, I do not see this command in Zarpli's project. Based on the information from the user manual, I assume that the endpoints /input/decode/on & off do not perform any actions. Rather, I think this is merely a status feedback from the sound system to the console, so the console correctly turns the decode LED on / off.

In short, I believe that the two endpoints /input/decode/on & off are of no use, but are intended for the console as feedback for the display of the decode LED. I suspect that this should rather be implemented as a read value endpoint, similar to the power state.

**Comment by LewisSmallwood** [comment]
Hi @3limin4tor, thank you for digging into this.

As you say, I think it should be a status accessor rather than for control.

I will take a further look and confirm the functionality.

Thank you

**Comment by LucianTothezan** [comment]
Are there any news on this project please?

**Comment by LucianTothezan** [comment]
hello again. if I may have one question please: if the TxRx pins were to be linked in parallel with the TxRx of the console, would the console react to the data sent from the esp module?(e.g.(if volume up from esp board, would the console`s leds light up?)
